### PR TITLE
Fix dependent-component versioning

### LINKED ISSUE #1832: Versioned component depending on axa-container does not render in pure html mode
Screenshot 2020-06-26 at 10 53 57

Code to reproduce:

```
<!DOCTYPE html>
<html lang="en">
  <head>
    <meta charset="UTF-8" />
    <meta name="viewport" content="width=device-width, initial-scale=1.0" />
    <meta http-equiv="X-UA-Compatible" content="ie=edge" />
    <title>Your awesome title</title>
  </head>
  <body>
    <axa-table-sortable
    datesortcolumnindex="3,4"
    innerscroll=""
    maxheight=""
    model='{"thead":[{"html":"Age","sort":"ASC"},{"html":"Name","sort":"ASC"},{"html":"City","sort":"DESC"},{"html":"Date of entry","sort":"DESC"},{"html":"Date of exit","sort":"DESC"},{"html":"Department"}],"tbody":[[{"html":"<span>55</span>"},{"html":"<span>Peter</span>"},{"html":"<span>Winterthur</span>"},{"html":"<span>22.04.2019</span>"},{"html":"<span>10.01.2020</span>"},{"html":"A"}],[{"html":"<span>22</span>"},{"html":"<span>Chris</span>"},{"html":"<span>Zürich</span>"},{"html":"<span>04.04.2019</span>"},{"html":"<span>04.06.2019</span>"},{"html":"B"}],[{"html":"<span>46</span>"},{"html":"<span>Hubert</span>"},{"html":"<span>Kreuzlingen</span>"},{"html":"<span>01.01.2020</span>"},{"html":"<span>05.02.2020</span>"},{"html":"C"}],[{"html":"<span>51</span>"},{"html":"<span>Petra</span>"},{"html":"<span>Kreuzlingen</span>"},{"html":"<span>08.10.2018</span>"},{"html":"<span>11.12.2019</span>"},{"html":"D"}],[{"html":"<span>18</span>"},{"html":"<span>Maria</span>"},{"html":"<span>Frauenfeld</span>"},{"html":"<span>29.04.2018</span>"},{"html":"<span>01.01.2020</span>"},{"html":"E"}],[{"html":"<span>29</span>"},{"html":"<span>David</span>"},{"html":"<span>Frauenfeld</span>"},{"html":"<span>02.05.2013</span>"},{"html":"<span>02.05.2015</span>"},{"html":"F"}]]}'
  ></axa-table-sortable>

    <script src="node_modules/@axa-ch/table-sortable/dist/index.js"></script>
  </body>
</html>
```

- create `index.html`
- `npm init` (spam ENTER)
- `npx http-server`

### PR DESCRIPTION
**Pull request by markus-walther**
Fixes #1832.

### CODE REVIEW

File: src/components/30-organisms/footer/index.js [hunk 1]
```diff
@@ -61,7 +61,7 @@ class AXAFooter extends InlineStyles {
     this.slotsNotPrepared = true;
 
     /* eslint-disable no-undef */
-    defineVersioned([AXAContainer], __VERSION_INFO__);
+    defineVersioned([AXAContainer], __VERSION_INFO__, this, this);
```
Comment (danyball) [inline src/components/30-organisms/footer/index.js]: Search-replace error. this is doubled

**Comment by danyball** [comment]
`const externalVersion = versionInfo[tagName];` --> versionInfo could be a string like "rsv". In this case something unexpected could happen.

**Comment by danyball** [comment]
`componentClass.versions = externalVersion;` --> why save this variable at `.versions` of the class? Have removed this logic for a test and everything works?! Without that we can remove many lines of code.

**Review by danyball — CHANGES_REQUESTED** [review]
I am going to test the code and give u feedback....

File: src/components/30-organisms/footer/index.js (hunk 1 above)
Comment (markus-walther) [reply src/components/30-organisms/footer/index.js]: Good catch!

**Comment by markus-walther** [comment]
> `const externalVersion = versionInfo[tagName];` --> versionInfo could be a string like "rsv". In this case something unexpected could happen.

('foo')['bar'] === undefined // true; welcome to the wonderfully bizarre world of JS... ;-)

**Comment by markus-walther** [comment]
> `componentClass.versions = externalVersion;` --> why save this variable at `.versions` of the class? Have removed this logic for a test and everything works?! Without that we can remove many lines of code.

This is done for live debugging of a website, see https://github.com/axa-ch/patterns-library/blob/develop/COMPONENT_VERSIONING.md, paragraph about `automatic npm version`.

**Comment by danyball** [comment]
> > `const externalVersion = versionInfo[tagName];` --> versionInfo could be a string like "rsv". In this case something unexpected could happen.
> 
> ('foo')['bar'] === undefined // true; welcome to the wonderfully bizarre world of JS... ;-)

Yes its working. But in this case its hard to understand why your code wants to select a key of a string. Its useless in this case. Why not add a if around?

**Comment by danyball** [comment]
> > `componentClass.versions = externalVersion;` --> why save this variable at `.versions` of the class? Have removed this logic for a test and everything works?! Without that we can remove many lines of code.
> 
> This is done for live debugging of a website, see https://github.com/axa-ch/patterns-library/blob/develop/COMPONENT_VERSIONING.md, paragraph about `automatic npm version`.

Ah ok for debugging. I would add a comment at that line. (i wondered why you are doing this)

**Comment by danyball** [comment]
It works - all my tests are working.

**Review by danyball — APPROVED** [review]

**Pull request merged**